.github/workflows/test.yml

```diff
@@ -96,14 +96,22 @@ jobs:
         echo "Repository status before tests: $git_status"
     - name: Show installed dependencies
       run: python -m pip list
-    - name: Run tests on Ubuntu
+    - name: Run tests on Ubuntu Test
       if: matrix.os == 'ubuntu-latest'
       run: |
         if [ ${{ matrix.code-cov }} ]; then codecov='--cov=openml --long  --cov-report=xml'; fi
         # Most of the time, running only the scikit-learn tests is sufficient
-        if [ ${{ matrix.sklearn-only }} = 'true' ]; then sklearn='-m sklearn'; fi
-        echo pytest -n 4 --durations=20 --dist load -sv $codecov $sklearn -o log_cli=true
-        pytest -n 4 --durations=20 --dist load -sv $codecov $sklearn -o log_cli=true
+        if [ ${{ matrix.sklearn-only }} = 'true' ]; then marks='sklearn and not production'; else marks='not production'; fi
+        echo pytest -n 4 --durations=20 --dist load -sv $codecov -o log_cli=true -m "$marks"
+        pytest -n 4 --durations=20 --dist load -sv $codecov -o log_cli=true -m "$marks"
+    - name: Run tests on Ubuntu Production
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        if [ ${{ matrix.code-cov }} ]; then codecov='--cov=openml --long  --cov-report=xml'; fi
+        # Most of the time, running only the scikit-learn tests is sufficient
+        if [ ${{ matrix.sklearn-only }} = 'true' ]; then marks='sklearn and production'; else marks='production'; fi
+        echo pytest -n 4 --durations=20 --dist load -sv $codecov -o log_cli=true -m "$marks"
+        pytest -n 4 --durations=20 --dist load -sv $codecov -o log_cli=true -m "$marks"
```

Comment:
This really should be parallelized in the matrix, but the special jobs make this very verbose, so breaking it up into two workflows _may_ be better. In any case we could do it as a separate PR?